src/torrent.cpp

```diff
@@ -7429,23 +7429,52 @@ bool is_downloading_state(int const st)
 		if (lhs->on_parole() != rhs->on_parole())
 			return lhs->on_parole();
 
-		// prefer to disconnect peers that send data at a lower rate
+		// prefer to disconnect peers that send data at a lower rate over those waiting to download
+		// the assumption is that a connection waiting to start downloading has the same potential 
+		// to be faster than the current slowest downloading connection than a new connection that 
+		// replaces a waiting connection - and disconnecting a waiting connection starts a new wait
+		// however ... to avoid torrents which wait forever there needs to be a maximum wait time
+		// after which a waiting connection is disconnected
 		std::int64_t lhs_transferred = lhs->statistics().total_payload_download();
 		std::int64_t rhs_transferred = rhs->statistics().total_payload_download();
+		bool const is_waiting = lhs_transferred * rhs_transferred == 0;
```

Comment:
I think this name is confusing. `is_waiting` sounds like it would refer to a peer, but this means "is one of the peer waiting". I also think there might be a better name referring to "connected by haven't sent data yet".

Reply:
This is set to true if one of the two connections being compared is waiting to download i.e. hasn't downloaded yet.

But we can certainly change it to something else if you can think of a better name.